src/python/pants/goal/run_tracker.py

```diff
@@ -390,3 +395,29 @@ def shutdown_worker_pool(self):
     N.B. This exists only for internal use and to afford for fork()-safe operation in pantsd.
     """
     SubprocPool.shutdown(self._aborted)
+
+  def report_target_info(self, scope, target, key, val):
+    """
+    Add target information to run_info under target_data
+    :param string scope: The running scope
+    :param string target: The target that we want to store info for
+    :param list of keys key: The key for the info being stored
+    :param dict or string val: The value of the info being stored
+    """
+    def insert_key_value(keys, val_item, index):
+      if index > 0:
+        new_val = {keys[index]: val_item}
+        insert_key_value(keys, new_val, index - 1)
+      else:
+        return {keys[index]: val_item}
+
+    val_to_store = insert_key_value(key, val, len(key) - 1)
+    target_data = self._target_data.get(target, None)
+    if target_data is None:
+      self._target_data.update({target: {scope: val_to_store}})
+    else:
+      scope_data = self._target_data[target].get(scope, None)
```

Comment:
This has different behavior than the current code - it won't stomp on all previous data from all scopes for a target, as the current code will, but it will stomp on any data for the (target, scope) pair that shares a common key prefix. Is that what we want?